book/animations.md

```diff
@@ -239,81 +288,28 @@ class Browser:
 says:[^virgl]
 
     OpenGL initialized: vendor=b'Red Hat', renderer=b'virgl'
 
-[^virgl]: In this case, `virgl` stands for "virtual GL", a way of
-hardware-accelerating the Linux subsystem of ChromeOS that works with the
-ChromeOS Linux sandbox.
-
-There are lots of resources online about how GPUs work and how to program them
-via GL shaders and so on. But we won't be writing shaders or other
-types of GPU programs in this book. Instead, let's focus on the basics of GPU
-technologies and how they map to browsers. A GPU is essentially a computer chip
-that is good at running very simple computer programs that specialize in
-turning simple data structures into pixels. These programs are so simple that
-the GPU can run one of them *in parallel* for each pixel, and this parallelism
-is why GPU raster is usually much faster than CPU raster. GPU draw is also
-much faster, because "copying" from one piece of GPU memory to another also
-happens in parallel for each pixel.
+[^virgl]: The `virgl` renderer stands for "virtual GL", a way of
+hardware-accelerating the Linux subsystem of ChromeOS that works with
+the ChromeOS Linux sandbox. This is a bit slower than using the GPU
+directly, so you'll probably see even faster raster and draw than I
+do.
 
-At a high level, the steps to raster and draw using the GPU are:[^gpu-variations]
+Now we can configure Skia to draw directly to the screen. The
+incantation is:[^weird]
 
-[^gpu-variations]: These steps vary a bit in the details by GPU architecture.
-
-* *Upload* the input data structures (that describe the display list)
-   to specialized GPU memory.
-
-* *Compile* GPU programs to run on the data structures.[^compiled-gpu]
-
-[^compiled-gpu]: That's right, GPU programs are dynamically compiled! This
-allows GPU programs to be portable across a wide variety of implementations
-that may have very different instruction sets or acceleration tactics.
-
-* For each desired surface, *execute* the raster into GPU textures.[^texture]
-
-[^texture]: A surface represented on the GPU is called a *texture*. There can be
-more than one surface, and practically speaking they often can't be rastered in
-parallel with each other.
-
-* *Draw* the textures onto the screen.
-
-You can configure any GPU program to draw directly to the *screen framebuffer*,
-or to an *intermediate texture*.[^gpu-texture] Draw proceeds bottom-up: the
-leaf textures are generated from raw display list data structures, and internal
-surface tree textures are generated by computing visual effects on one or more
-children. The root of the tree is the framebuffer texture.
-
-[^gpu-texture]: Recall from [Chapter 11](visual-effects.md) that there can be
-surfaces that draw into other surfaces, forming a tree. Skia internally does
-this sometimes, based on various triggers such as blend modes. The internal
-nodes draw into intermediate textures.
-
-The time to run GPU raster is roughly the sum of the time for these four steps.
```

Comment:
I think this paragraph is good to keep, as it explains how slowdowns can come from each part. I suppose we don't actually use this knowledge though. Footnote?

Reply:
I merged this text into the `example-cpu-fast` footnote.